File: backend/app_test/shared/test_assets/uploads/generate.py
from typing import Any, Dict, Optional


def generate() -> Dict[str, Any]:
    a = 1
    b = 2
    sum_val = a + b
    print("This is the value of a", a)
    print("This is the value of b", b)
    print("This is a structure", {"params": {"a": a, "b": b}})

    return {
        "params": {"a": a, "b": b},
        "correct_answers": {"sum": sum_val},
        "intermediate": {"step": f"{a} + {b} = {sum_val}"},
        "test_results": {"pass": 1, "message": "Addition successful"},
        "nDigits": 3,
        "sigfigs": 3,
    }

File: backend/app_test/shared/test_assets/uploads/generate.js


function generate() {
  const a = 1;
  const b = 2;
  const sum = a + b;

  console.log("This is the value of a", a);
  console.log("This is the value of b", b);
  data = { params: { a: a, b: b } };
  console.log("Here is a structure", data);

  return {
    params: { a: a, b: b },
    correct_answers: { sum: sum },
    intermediate: { step: `${a} + ${b} = ${sum}` },
    test_results: { pass: 1, message: "Addition successful" },
    nDigits: 3,
    sigfigs: 3,
  };
}
